pkg/services/gitops/install.go

```diff
@@ -229,3 +259,47 @@ func (g *Gitops) applyManifestsToK8s(ctx context.Context, namespace string, mani
 
 	return nil
 }
+
+const LabelPartOf = "app.kubernetes.io/part-of"
+
+var ErrNamespaceNotFound = errors.New("namespace not found")
+
+func (g *Gitops) fetchNamespaceWithLabel(ctx context.Context, key string, value string) (string, error) {
+	selector := labels.NewSelector()
+
+	partOf, err := labels.NewRequirement(key, selection.Equals, []string{value})
+	if err != nil {
+		return "", fmt.Errorf("bad requirement: %w", err)
+	}
+
+	selector = selector.Add(*partOf)
+
+	options := client.ListOptions{
+		LabelSelector: selector,
+	}
+
+	nsl := &corev1.NamespaceList{}
+	if err := g.kube.Raw().List(ctx, nsl, &options); err != nil {
+		return "", fmt.Errorf("error setting resource: %w", err)
+	}
+
+	if len(nsl.Items) == 0 {
+		return "", ErrNamespaceNotFound
+	}
+
+	return nsl.Items[0].Name, nil
+}
+
+func (g *Gitops) saveWegoConfig(ctx context.Context, params InstallParams) error {
+	fluxNamespace, err := g.fetchNamespaceWithLabel(ctx, LabelPartOf, "flux")
+	if err != nil {
+		if !errors.Is(err, ErrNamespaceNotFound) {
```

Comment:
It looks like you are dropping the `ErrNamespaceNotFound` error here.  What about?

```
if errors.Is(err, ErrNamespaceNotFound) {
  return err
else if err != nil {
 return fmt.Errorf("failed fetching flux namespace: %w", err)
}
```
Also can you write some unit tests capturing this scenario when `Install` is called?